File: backend/internal/model/monitoring_point.go
package model

import "time"

type MonitoringPoint struct {
	ID                    uint      `gorm:"primaryKey"`
	PointCode             string    `gorm:"size:64;uniqueIndex;not null"`
	Name                  string    `gorm:"size:160;not null"`
	XM                    float64   `gorm:"not null"`
	YM                    float64   `gorm:"not null"`
	HeightM               float64   `gorm:"not null"`
	AreaType              string    `gorm:"size:40;not null"`
	BackgroundProfileJSON string    `gorm:"type:text;not null"`
	OwnerTeam             string    `gorm:"size:120;not null"`
	PointState            string    `gorm:"size:24;index;not null"`
	Version               uint      `gorm:"not null;default:1"`
	CreatedAt             time.Time `gorm:"not null"`
	UpdatedAt             time.Time `gorm:"not null"`
}

func (MonitoringPoint) TableName() string { return "monitoring_points" }

File: frontend/src/types/monitoring-point.ts
import type { Spectrum } from './common'

export interface MonitoringPoint {
  id: number
  point_code: string
  name: string
  x_m: number
  y_m: number
  height_m: number
  area_type: 'boundary' | 'workshop' | 'office' | 'residential'
  background_profile: Spectrum
  owner_team: string
  point_state: 'active' | 'inactive'
  version: number
  measurement_count: number
  latest_quality: string
  latest_measurement_time: string | null
  created_at: string
  updated_at: string
}

export interface CreateMonitoringPoint {
  point_code: string
  name: string
  x_m: number
  y_m: number
  height_m: number
  area_type: MonitoringPoint['area_type']
  background_profile: Spectrum
  owner_team: string
}
